catalog/dags/common/storage/media.py

```diff
@@ -274,26 +281,28 @@ def _enrich_meta_data(meta_data, license_url, raw_license_url) -> dict:
             )
         return enriched_meta_data
 
-    def _enrich_tags(self, raw_tags) -> list | None:
+    def _enrich_tags(self, raw_tags: set | None) -> list | None:
         """
-        Add provider information to tags.
+        Add provider information to tags while filtering out denied terms.
 
         Args:
-            raw_tags: List of strings or dictionaries
+            raw_tags: Set of strings
 
         Returns:
             A list of 'enriched' tags:
             {"name": "tag_name", "provider": self._PROVIDER}
         """
-        if not isinstance(raw_tags, list):
-            logger.debug("`tags` is not a list.")
-            return None
-        else:
-            return [
-                self._format_raw_tag(tag)
-                for tag in raw_tags
-                if not self._tag_blacklisted(tag)
-            ]
+        if not raw_tags:
+            return
+
+        if not isinstance(raw_tags, set):
```

Comment:
Instead of enforcing a `set` here (and having to change the logic in every provider), would it be easiest to just cast a `list` to `set` to remove the duplicates at this step while also accepting `set` as a valid type being passed into this function?

Reply:
I was thinking of that option, yes, that would have been easier but at the same time, we still have a reasonably number of providers to allow a change like this so I went a step further and applied it to all the providers collecting tags. I think reducing fields to one type makes things easier in the long run, helping remember what kind of data we want here.

Reply:
To minimize changes, I would rather keep the tags as a list of strings in the provider scripts, and convert them to a set here, in the `MediaStore` class.

I think that most of the duplicates were not supplied by the providers, but instead created by us when we incorrectly decoded the tags and saved both versions, or lower-cased the tags, and then saved both versions.

Reply:
> we could very easily enforce that constraint in the MediaStore class

That is what the PR is doing.

I don't see the complexity in the change to a set; we have made more prominent changes in the past. As I said before, defining the type of fields clearly helps to decide what data goes in and it's easier to manage one data type instead of two. Also, sets aren't really more complex than a list.